### ISSUE TITLE
minimum, maximum and extrema with specifing dims for an array that have all missing value dim doesn't work 

### ISSUE THREAD

**Issue by Paalon** [JuliaLang/julia#35308]
`minimum`, `maximum` and `extrema` with specifing `dims` for an array that have all missing value dim doesn't work.

```julia
julia> minimum([1 missing; 2 missing], dims=1)
ERROR: TypeError: non-boolean (Missing) used in boolean context
Stacktrace:
 [1] reducedim_init(::Function, ::typeof(min), ::Array{Union{Missing, Int64},2}, ::Int64) at ./reducedim.jl:148
 [2] _mapreduce_dim(::Function, ::Function, ::NamedTuple{(),Tuple{}}, ::Array{Union{Missing, Int64},2}, ::Int64) at ./reducedim.jl:317
 [3] #mapreduce#580 at ./reducedim.jl:307 [inlined]
 [4] _minimum at ./reducedim.jl:679 [inlined]
 [5] _minimum at ./reducedim.jl:678 [inlined]
 [6] #minimum#589 at ./reducedim.jl:652 [inlined]
 [7] top-level scope at REPL[6]:1

```

**Comment by pdeffebach** [comment]
Note that

```
julia> minimum([1, 2, missing])
missing

julia> min(1, missing)
missing
```

So this is *just* about the keyword argument for `dims`. This is not a case of adding `missing` propagation to a function that otherwise errors.

**Comment by nalimilan** [comment]
Good catch. I have an inefficient fix at https://github.com/JuliaLang/julia/pull/35323. An efficient one would take much more work.

**Comment by stev47** [comment]
seems to be fixed by 76952a88ea650ae8c6b6b1d010ef695ed9c8244d

**Comment by nalimilan** [comment]
As noted at https://github.com/JuliaLang/julia/pull/35323, this isn't completely fixed unfortunately. See the note there:
```
                # TODO: Some types, like BigInt, don't support typemin/typemax.
                # So a Matrix{Union{BigInt, Missing}} can still error here.
```
